book/src/exec.md

```diff
@@ -0,0 +1,55 @@
+## On a high level
+Users use intents when they want to exchange their assets for some other asset. High level description of the flow:
+1. **Create**: User creates an intent publishing the information of what they have and what they want to get in exchange
+2. **Gossip**: The intent goes to the intent gossip network and gossiped around the solver nodes.
+3. **Solve**: Solvers find matching intents and update the intent until it is fully satisfied
+4. **Finalize**: When the intent is satisfied, a transaction is created and published on the blockchain
+
+![img_1.png](img/img_exec_high.png)
+
+### Create an intent
+User intents can be seen as partial transactions. Users send them to the intent gossip network and solvers match them in order to create full transactions and publish them on the blockchain. That implies that users need to give some information to the solvers that is sufficient to create the final transaction (proofs). To make sure that users reveal the minimal amount of private infortmation that is still enough to create the transaction, users create **intermediate notes** from the notes they are willing to spend.
+
+#### Intermediate notes
+An intermediate note is a note with a simplified sending VP. This VP is specific to the intent and contains only the rules that are necessary to satisfy for the transaction to happen. All of the note fields stay the same except:
```

Comment:
How is the VP simplifed in an intermediate note?

Reply:
It includes only the rules governing the specific pieces of state in question, and those pieces of state are sent from the user VP to this temporary VP

e.g. if Alice wants to trade A for B, Alice would first send A from her user VP to a temporary VP stipulating that A can be sent to anyone who sends her B

it'd be good to keep a running example through these notes